addons/account/models/account_move.py

```diff
@@ -2415,29 +2415,31 @@ def _apply_cash_rounding(self, diff_balance, diff_amount_currency, cash_rounding
 
     @contextmanager
     def _sync_unbalanced_lines(self, container):
+        def has_tax(move):
+            return bool(move.line_ids.tax_ids)
+
+        move_had_tax = {move: has_tax(move) for move in container['records']}
         yield
+        move_has_tax = {move: has_tax(move) for move in container['records']}
```

Comment:
@william-andre This dict is not needed, `has_tax` can be directly called after the yield when needed